### PR TITLE
feat: lambda provider

### PR DESCRIPTION
**Pull request by b4s36t4**
**Title:** 
- Adds lambda chat as new provider to gateway.

**Related Issues:** (optional)
- #650

### CODE REVIEW

File: src/providers/lambda/api.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { ProviderAPIConfig } from '../types';
+
+export const LambdaAPIConfig: ProviderAPIConfig = {
+  getBaseURL: () => `https://api.lambdalabs.com/v1`,
+  headers({ providerOptions }) {
+    const { apiKey } = providerOptions;
+    const headers = {
+      Authorization: apiKey,
```
Comment (VisargD) [inline src/providers/lambda/api.ts]: As per their docs examples, `Bearer` is required in the `Authorization` header. Can we please add it here as well?

Comment (VisargD) [reply src/providers/lambda/api.ts]: https://docs.lambdalabs.com/on-demand-cloud/using-the-lambda-chat-completions-api#creating-chat-completions

File: src/providers/lambda/index.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  chatCompleteParams,
+  completeParams,
+  responseTransformers,
+} from '../open-ai-base';
+import { ProviderConfigs } from '../types';
+import { LambdaAPIConfig } from './api';
+
+export const LambdaProviderConfig: ProviderConfigs = {
+  chatComplete: chatCompleteParams([], { model: 'Liquid-AI-40B' }),
+  complete: completeParams([], { model: 'Liquid-AI-40B' }),
```
Comment (VisargD) [inline src/providers/lambda/index.ts:12]: Is this the correct model ID? Their docs does not seem to mention this exact id anywhere. So I am not sure.

Comment (b4s36t4) [reply src/providers/lambda/index.ts:12]: I took it from their playground. They do support the `liquid` models as well.

Comment (VisargD) [reply src/providers/lambda/index.ts:12]: Got it

**Review by VisargD — APPROVED** [review]

**Pull request merged**